src/util/bip32.rs

```diff
@@ -572,7 +644,50 @@ impl ExtendedPubKey {
         })
     }
 
-    /// Returns the HASH160 of the public key of the xpub
+    /// Decoding extended public key from binary data according to BIP 32
+    pub fn decode(data: &[u8]) -> Result<ExtendedPubKey, Error> {
+        if data.len() != 78 {
+            return Err(Error::WrongExtendedKeyLength(data.len()))
+        }
+
+        let cn_int: u32 = endian::slice_to_u32_be(&data[9..13]);
+        let child_number: ChildNumber = ChildNumber::from(cn_int);
```

Comment:
nit: Type doesn't need to be specified twice.

Reply:
True, remnants on the older code